packages/dcos-integration-test/extra/test_mesos.py

```diff
@@ -157,7 +158,11 @@ def test_if_ucr_app_runs_in_new_pid_namespace(dcos_api_session):
         marathon_framework_id = dcos_api_session.marathon.get('/v2/info').json()['frameworkId']
         app_task = dcos_api_session.marathon.get('/v2/apps/{}/tasks'.format(app['id'])).json()['tasks'][0]
 
-        content = dcos_api_session.mesos_sandbox_file(
-            app_task['slaveId'], marathon_framework_id, app_task['id'], ps_output_file)
+        # There is a race between the `app_task` starting and it writing its output to the `ps_output_file`.
+        # We wait up to 10 seconds for this file to appear before giving up and throwing an exception.
+        @retrying.retry(wait_fixed=0.1, stop_max_delay=10000, retry_on_exception=lambda x: True)
```

Comment:
You could avoid retrying and polling if you had the task notify the test via network call. Ex: curl from the task to a server fixture. This might make the test more responsive.